source/tester/tester-tests.adb

```diff
@@ -155,6 +159,96 @@ package body Tester.Tests is
       end loop;
    end Do_Send;
 
+   procedure Do_Shell
+     (Self    : in out Test'Class;
+      Command : GNATCOLL.JSON.JSON_Value)
+   is
+      pragma Unreferenced (Self);
+
+      function To_Program (Name : String) return String;
+      --  Take base name of the command and find it on PATH
+
+      function To_Program (Name : String) return String is
+         Found : GNAT.OS_Lib.String_Access :=
+           GNAT.OS_Lib.Locate_Exec_On_Path (Name);
+      begin
+         return Result : constant String := Found.all do
+            Free (Found);
+         end return;
+      end To_Program;
+
+      List  : constant GNATCOLL.JSON.JSON_Array := Command.Get;
+      Cmd   : constant String := To_Program (GNATCOLL.JSON.Get (List, 1).Get);
```

Comment:
let's try running $SHELL if it exists with the given command line, rather than read a program from the list? if people want to run something more complex than a shell script, they can use a wrapper.

Alternatively we can decide to only support launching a Python script, perhaps more portable.

Reply:
We can easily run the Python (or any $SHELL) with this "shell" command. In the test I replaced the shell command with Python and added it to the README.md as example

        "shell": ["${PYTHON}", "${DIR}/makelink.py" ]

Is it good enough?